### DISCUSSION TITLE
Custom non-time-based triggers?

### DISCUSSION THREAD

**Discussion by backfromexile** [quartznet/quartznet#2321] [Q&A]
As far as I can see, Quartz.NET only provides an API for time-based triggers like cron-based or interval-based ones.  
What exactly do we need to do to add custom non-time-based triggers? For example, we would like to implement a trigger that fires using an underlying `FileSystemWatcher` that listens to new files or file changes.
Custom persistence logic would be nice as well, but is not a priority or requirement, but the triggers have to work in memory if persistence is not as easy.

**Comment by lahma** [comment]
You might want to check the plugins, there's already plugin that monitors file system for changes and then does custom logic. So this is more like polling trigger.

https://github.com/quartznet/quartznet/tree/main/src/Quartz.Jobs/Job

**Comment by backfromexile** [comment]
Hi, this doesn't really answer my question.

Sure, we know that we could just register a job that triggers every x seconds, but we want to avoid that if possible.
Also we don't just have filesystem-based custom triggers, but also other ones that ideally trigger based on an event. The FileSystemWatcher was just an example here.

**Reply by jafin** [reply]
Have you investigated other pub/sub libs ?  This sounds more aligned to a event publisher/subscriber model..   Not sure I see what benefit quartz may provide over this?  Given Quartz is very time/scheduler oriented.

**Comment by backfromexile** [comment]
We did and unfortunately other similar  job scheduling frameworks suffer from the same issue.
We *do* have cron/interval-triggered jobs, but we also have jobs that are triggered by other non-time-based events. Sure, it's possible here as well by scheduling the job(s) whenever such an event is fired, but it feels wrong especially when also using the cron or interval-based triggers that Quartz provides.
Ideally we'd love to use an existing solution, but at the moment it looks like we will use one library as a base and build a custom solution around it.
